runtime/src/bank/check_transactions.rs

```diff
@@ -97,18 +97,25 @@ impl Bank {
             .zip(lock_results)
             .map(|(tx, lock_res)| match lock_res {
                 Ok(()) => {
-                    let compute_budget_limits = process_compute_budget_instructions(
+                    let compute_budget_and_limits = process_compute_budget_instructions(
                         tx.borrow().program_instructions_iter(),
                         &self.feature_set,
-                    );
+                    )
+                    .map(|limit| {
+                        if let Some(compute_budget) = self.compute_budget {
+                            limit.get_limits_using_compute_budget(compute_budget)
+                        } else {
+                            limit.get_compute_budget_and_limits()
+                        }
+                    });
```

Comment:
this changes logic; why favor bank.compute_budget over the one fetched from transaction?

Reply:
The logic hasn't changed actually. See: https://github.com/anza-xyz/agave/blob/1bcf743f445270407bccf55681f74e06ef8b9a48/svm/src/transaction_processor.rs#L944

It should be streamlined though. It'll need a feature gate. With this PR, it's moved out of SVM, and can be reworked within monorepo.